test/transportSpec.js

```diff
@@ -248,13 +295,13 @@ describe('Transport', function () {
       yield this.fulfilled
     })
 
-    it('should not fulfill when it generates the wrong fulfillment', function * () {
+    it('should reject when it generates the wrong fulfillment', function * () {
```

Comment:
It would also be good to include a test to make sure that when the packet is modified slightly it will fail. It's kind of similar to this one but in this case the execution condition isn't even valid